### PR TITLE
Make ghci parts reusable from cunumeric

### PR DESCRIPTION
**Pull request by sandeepd-nv**

### CODE REVIEW

File: .github/workflows/gh-build.yml [hunk 1]
```diff
@@ -43,44 +44,52 @@ jobs:
...
-            --build-arg AWS_SECRET_ACCESS_KEY=${{ env.AWS_SECRET_ACCESS_KEY }} \
-            --build-arg GITHUB_TOKEN=${{ env.GITHUB_TOKEN }} \
-            --build-arg USE_CUDA=${{ env.USE_CUDA }} \
-            --progress=plain \
-            --tag=$IMAGE_NAME:${{ inputs.sha }} \
-            --label "git-commit=${{ inputs.sha }}" \
-            -f continuous_integration/Dockerfile .
+          IMAGE_TAG=${{ env.IMAGE_NAME }}:${{ inputs.sha }}
+
+          chmod +x continuous_integration/build-docker-image
```
Comment (marcinz) [inline .github/workflows/gh-build.yml:59]: Should the permission just be changed in the repository?

File: CMakeLists.txt [hunk 2]
```diff
@@ -72,7 +72,7 @@ include(rapids-cuda)
 include(rapids-export)
 include(rapids-find)
 
-set(legate_core_version 23.07.00)
+set(legate_core_version 23.09.00)
```
Comment (marcinz) [inline CMakeLists.txt:75]: This should come from a merge of 23.09?

Comment (marcinz) [reply CMakeLists.txt:75]: Or is it something I forgot?

File: continuous_integration/Dockerfile [hunk 3]
```diff
@@ -28,16 +30,22 @@ COPY --chown=coder:coder . /home/coder/legate
 
 RUN chmod a+x /home/coder/.local/bin/* && \
     mkdir -p /tmp/out && \
-    chown -R coder:coder /tmp/out  && \
-    chown -R coder:coder /home/coder/.artifacts
+    chown -R coder:coder /tmp/out && \
+    sudo chown -R coder:coder /home/coder
```
Comment (marcinz) [inline continuous_integration/Dockerfile:34]: Why is that necessary? Are the permissions wrong to begin with?

File: continuous_integration/home/coder/.local/bin/entrypoint [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+    eval "export $(find /run/secrets/ -type f -exec bash -c 'echo ${0/\/run\/secrets\//}=$(<${0})' {} \;)";
+    if [ "${history_enabled}" -eq "0" ]; then { set -o history; } 2>/dev/null; fi;
+    if [ "${xtrace_enabled}" -eq "0" ]; then { set -o xtrace; } 2>/dev/null; fi;
+
+    . devcontainer-utils-post-attach-command;
 
+    sleep 10;
+    . devcontainer-utils-vault-s3-test;
+    . devcontainer-utils-vault-s3-export 0;
+}
```
Comment (marcinz) [inline continuous_integration/home/coder/.local/bin/entrypoint:21]: My understanding of this is lacking. :) Why do we need this?

File: .github/workflows/gh-build.yml [hunk 5]
```diff
@@ -43,44 +44,52 @@ jobs:
           role-duration-seconds: 28800 # 8 hours
           role-to-assume: arn:aws:iam::279114543810:role/gha-oidc-nv-legate
 
+      - name: Docker system prune
+        run: |
+          docker version
+          docker system prune --all --force
```
[the remark this answers is not in the document]
Comment (sandeepd-nv) [reply .github/workflows/gh-build.yml:50]: Yes. The build fails due to insufficient space sometimes.

File: .github/workflows/gh-build.yml [hunk 6]
```diff
@@ -15,6 +15,7 @@ on:
 
 env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  BASE_IMAGE: rapidsai/devcontainers:23.06-cpp-cuda11.8-mambaforge-ubuntu22.04
```
[the remark this answers is not in the document]
Comment (sandeepd-nv) [reply .github/workflows/gh-build.yml:18]: Let's do it in a separate PR.

File: continuous_integration/Dockerfile (hunk 3 above)
Comment (sandeepd-nv) [reply continuous_integration/Dockerfile:34]: The ~/.artifacts folder gets mounted with root as owner. This simply changes the ownership back to coder.

File: continuous_integration/home/coder/.local/bin/entrypoint (hunk 4 above)
Comment (sandeepd-nv) [reply continuous_integration/home/coder/.local/bin/entrypoint:21]: > devcontainer-utils-vault-s3-test

This tests if AWS credentials are valid.

> devcontainer-utils-vault-s3-export

This sets up various SCCACHE_* env vars through ~/.aws/config or ~/.bashrc.

This things are required for enabling SCCACHE for "local repro" where AWS auth happens using a GH personal access token.

File: .github/workflows/gh-build.yml (hunk 1 above)
Comment (sandeepd-nv) [reply .github/workflows/gh-build.yml]: Done.

File: CMakeLists.txt (hunk 2 above)
Comment (sandeepd-nv) [reply CMakeLists.txt]: This change was made before the default branch was switched to 23.09. I have rebased my code now.

File: continuous_integration/Dockerfile (hunk 3 above)
Comment (sandeepd-nv) [reply continuous_integration/Dockerfile]: Line removed.

File: .github/workflows/gh-build.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (m3vaz) [reply .github/workflows/gh-build.yml:18]: Separate PR would be ideal. Keeping conversation open so we don't forget to do this later.

File: continuous_integration/home/coder/.local/bin/entrypoint (hunk 4 above)
Comment (m3vaz) [reply continuous_integration/home/coder/.local/bin/entrypoint:21]: I would also add that the 10 second wait is essential. The credentials that we get are generated on the fly, and need to be uploaded to AWS. We hit authentication issues if we try to race against the upload process.

File: continuous_integration/home/coder/.local/bin/conda-utils [hunk 7]
```diff
@@ -58,10 +58,12 @@ get_yaml_and_make_conda_env() {
     make_conda_env_from_yaml;
 }
 
-make_conda_env_using_legate_core() {
-    mamba create -n "${DEFAULT_CONDA_ENV:-legate}"
+install_legate_core_with_war() {
```
Comment (m3vaz) [inline continuous_integration/home/coder/.local/bin/conda-utils:61]: Can you file a GH issue describing the issue and then reference the issue # here?

**Review by m3vaz — APPROVED** [review]
We should file an issue on github tracking the numpy discrepancy (i.e. the reason why we need the WAR) and reference the issue number in the WAR comment, but otherwise LGTM

File: continuous_integration/home/coder/.local/bin/conda-utils (hunk 7 above)
Comment (sandeepd-nv) [reply continuous_integration/home/coder/.local/bin/conda-utils:61]: Done. See #812.

File: .github/workflows/gh-build.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (trxcllnt) [reply .github/workflows/gh-build.yml:18]: I did this in https://github.com/nv-legate/legate.core/pull/725

File: continuous_integration/home/coder/.local/bin/entrypoint (hunk 4 above)
Comment (trxcllnt) [reply continuous_integration/home/coder/.local/bin/entrypoint:21]: This is also done in the 23.08 devcontainers (and removed in https://github.com/nv-legate/legate.core/pull/725).
